src/base/_foundation.lua

```diff
@@ -244,9 +244,13 @@
 
 		local res = os.locate(fname, with_ext, p5, p4)
 		res = res or fname
-		local compiled_chunk = loadfile(res)
+		local compiled_chunk, err = loadfile(res)
 		if compiled_chunk == nil then
-			premake.error("Cannot find either " .. table.implode({fname, with_ext, p5, p4}, "", "", " or "))
+			if err then
```

Comment:
I was going by the existing usage in the codebase, where premake.main.moduleLoader() does this:
```lua
				local chunk, err = loadfile(file)
				if chunk then
					return chunk
				end
				if err then
					return "\n\tload error " .. err
				end
```
Which is effectively what I have. Happy to change it, as I thought the same thing when I was writing it, but opted to stick with consistent usage since I'm not very familiar with lua.

Reply:
I can't really comment on why `moduleLoader` is like that, but my understanding of `loadfile` is that it returns `nil, err` when there's an error, otherwise it just returns a function. So, I don't think there's a scenario where `nil, nil` would occur - you can validate this by attempting to load a file that you know doesn't exist, since that's what the error in the `else` path is for.

However, I think the block you've modified could probably be changed to this:
```lua
local res = os.locate(fname, with_ext, p5, p4)
if res == nil then
  premake.error("Cannot find either " .. table.implode({fname, with_ext, p5, p4}, "", "", " or "))
end

local compiled_chunk, err = loadfile(res)
if compiled_chunk == nil then
  premake.error("Cannot load " .. res .. ": " .. err)
end
```

Note, I changed `fname` to `res` in the error message you added as it will tell the user exactly which file we loaded instead of which `include` failed. For example, if a user has both `x.lua` and `x/premake5.lua` and does `include "x"`, Premake will load `x.lua` while the user may be expecting `x/premake5.lua` to be loaded.

Reply:
@samsinsane : Thanks for the feedback - I've updated the PR!

I took most of your recommendations, except I still don't display `res` in the error message, as it included in `err` in every case I tested (with the addition of line and error specifics). The original `fname` that is passed to the function is displayed instead still, as that's useful since it's what the users script contains. I also added information about the caller to the output, so you know where the include originated from.

In my testing I found another area that errors weren't being usefully caught/displayed for includes, this time when the code is being executed. I've wrapped that in pcall ('protected call', lua's equivalent of try/catch) and displayed a similarly detailed error message.

Example output:

Missing file included:
```console
Error: ** Error: D:\dev\build-sys-eval\foobar\src\foobar\L1a\premake5.lua(30): Error executing 'tools/premake/badcode.lua: ** Error: D:\dev\build-sys-eval\foobar\src\tools\premake\badcode.lua(56): Cannot find neither no-file-with-this-name nor no-file-with-this-name.lua nor no-file-with-this-name/premake5.lua nor no-file-with-this-name/premake4.lua
```

Syntax error in included file:
```console
Error: ** Error: D:\dev\build-sys-eval\foobar\src\foobar\L1a\premake5.lua(30): Error loading 'tools/premake/badcode.lua: D:/dev/build-sys-eval/foobar/src/tools/premake/badcode.lua:57: syntax error near ']'
```

Broken include of a premake-internal script:
```console
Error: ** Error: D:\dev\build-sys-eval\foobar\src\foobar\L1a\premake5.lua(30): Error executing 'tools/premake/badcode.lua: ** Error: D:\dev\build-sys-eval\foobar\src\tools\premake\badcode.lua(59): Error executing '$/self-test/test_declare.lua: [string "self-test/test_declare.lua"]:6: attempt to index a nil value (local 'm')
```